### PR TITLE
[Merged by Bors] - feat: port RingTheory.Localization.FractionRing

### PR DESCRIPTION
**Pull request by mpenciak**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Localization/FractionRing.lean [hunk 1]
```diff
@@ -300,27 +298,14 @@ def FractionRing :=
...
-    add := (· + ·)
-    mul := (· * ·)
-    neg := Neg.neg
-    sub := Sub.sub
-    one := 1
-    zero := 0
-    nsmul := AddMonoid.nsmul
-    zsmul := SubNegMonoid.zsmul
-    npow := Localization.npow _ }
+noncomputable instance : Field (FractionRing A) := IsFractionRing.toField A
```
Comment (jcommelin) [inline Mathlib/RingTheory/Localization/FractionRing.lean:308]: Is there a reason why this golf wasn't used in ML3?

Comment (Parcly-Taxel) [reply Mathlib/RingTheory/Localization/FractionRing.lean:310]: I've added a porting note explaining why this instance was golfed.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/RingTheory/Localization/FractionRing.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/RingTheory/Localization/FractionRing.lean:310]: There was no reason. ML3 builds fine with this change. I will assume we don't want to backport it.

**Pull request closed without merging**